examples/azure/terraform-azure-availabilityset-example/main.tf

```diff
@@ -0,0 +1,124 @@
+provider "azurerm" {
+  version = "=2.20.0"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# See test/azure/terraform_azure_availabilityset_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
```

Comment:
NIT: This block should be at the top of the file.

Reply:
Fixed all occurrences. I updated the other Azure main.tf files as well with a header format that combines the recently added description and the test instructions for consistency.